### PR TITLE
storage controller: use 'lazy' mode for location_config

### PR DESCRIPTION
**Pull request by jcsp**
## Problem

If large numbers of shards are attached to a pageserver concurrently, for example after another node fails, it can cause excessive I/O queue depths due to all the newly attached shards trying to calculate logical sizes concurrently.

#6907 added the `lazy` flag to handle this.

## Summary of changes

- Use `lazy=true` from all /location_config calls in the storage controller Reconciler.

## Checklist before requesting a review

- [ ] I have performed a self-review of my code.
- [ ] If it is a core feature, I have added thorough tests.
- [ ] Do we need to implement analytics? if so did you add the relevant metrics to the dashboard?
- [ ] If this PR requires public announcement, mark it with /release-notes label and add several sentences in this section.

## Checklist before merging

- [ ] Do not forget to reformat commit message to not include the above checklist

### CODE REVIEW

File: control_plane/attachment_service/src/reconciler.rs [hunk 1]
```diff
@@ -114,11 +114,16 @@ impl Reconciler {
             .locations
             .insert(node.id, ObservedStateLocation { conf: None });
 
+        // 'lazy' mode means that a tenant's activiation will be subject to a concurrency limit
+        // in the pageserver: this is the safer way to attach a tenant, as it avoids thundering
+        // herd I/O when attaching many tenants at the same time.
+        let lazy = true;
```
Comment (koivunej) [inline control_plane/attachment_service/src/reconciler.rs:120]: If I understand correctly, `Reconciler::live_migrate` will call with lazy=true as well and as we discussed just now (re: #7000) we should not do it.

I see two alternatives:
- complicate calling this fn
- always use SpawnMode::Eager on pageserver side

File: pageserver/client/src/mgmt_api.rs [hunk 2]
```diff
@@ -251,17 +251,18 @@ impl Client {
...
-            "{}/v1/tenant/{}/location_config",
+            "{}/v1/tenant/{}/location_config?lazy={lazy}",
             self.mgmt_api_endpoint, tenant_shard_id
         );
         let path = if let Some(flush_ms) = flush_ms {
-            format!("{}?flush_ms={}", path, flush_ms.as_millis())
+            format!("{}&flush_ms={}", path, flush_ms.as_millis())
         } else {
             path
         };
```
Comment (koivunej) [inline pageserver/client/src/mgmt_api.rs:268]: reqwest has a https://docs.rs/reqwest/latest/reqwest/struct.Url.html#method.query_pairs_mut for this not forcing to remember there's already one.

**Review by koivunej — APPROVED** [review]
Without the live migration lazy=true issue this is looking fine.

File: control_plane/attachment_service/src/reconciler.rs (hunk 1 above)
Comment (jcsp) [reply control_plane/attachment_service/src/reconciler.rs]: Yep, it's easy enough to make use of lazy=true specific to the case where we want it, so I've just made this an argument to the function, and set it true for the "general case" part of Reconciler::reconcile (and false for live migrations)

File: pageserver/client/src/mgmt_api.rs (hunk 2 above)
Comment (jcsp) [reply pageserver/client/src/mgmt_api.rs]: Nice, amended to use that.

**Pull request merged**